enhancements/monitoring/optional-monitoring-capability.md

```diff
@@ -0,0 +1,994 @@
+
+<!--
+Are there any unique considerations for making this change work with
+Hypershift?
+
+See https://github.com/openshift/enhancements/blob/e044f84e9b2bafa600e6c24e35d226463c2308a5/enhancements/multi-arch/heterogeneous-architecture-clusters.md?plain=1#L282
+
+How does it affect any of the components running in the
+management cluster? How does it affect any components running split
+between the management cluster and guest cluster?
+-->
+
+Disabling the capability translates to hypershift clusters pointing
+[`METRICS_SET` environment variable] to `Telemetry`, in order to
+minimize the monitoring footprint while ensuring that telemetry
+operations are not affected.
+
+[`METRICS_SET` environment variable]: https://github.com/openshift/hypershift/blob/9e76b0a736a85e29fd69a76bca2b2968aa8db8b9/hack/dev/aws/run-operator-locally-aws-dev.sh#L4
+
+#### Standalone Clusters
+
+<!--
+Is the change relevant for standalone clusters?
+-->
+
+For standalone clusters, the behavior is the same as described in the
+**Workflow Description** section.
+
+#### Single-node Deployments or MicroShift
+
+<!--
+How does this proposal affect the resource consumption of a
+single-node OpenShift deployment (SNO), CPU and memory?
+
+How does this proposal affect MicroShift? For example, if the proposal
+adds configuration options through API resources, should any of those
+behaviors also be exposed to MicroShift admins through the
+configuration file for MicroShift?
+-->
+
+Since MicroShift does not directly rely on the components deployed
+by CMO for its monitoring needs (for e.g., [metrics-server]), the
+capability will not have any direct effect on MicroShift deployments.
+MicroShift [remote-writes] it's telemetry data to the `metrics/v1/receive`
+endpoint, where the rules are validated against the [telemetry
+whitelist], in addition to other operations. Note that these rules
+are the ones used to draw its dashboards, which is limited to
+[grafana-dashboard-microshift-telemetry] at this time.
+
+#### NVIDIA DOCA Platform Framework-managed clusters
+
+DPF deployments use a dual-cluster architecture: a management cluster (x86) that
+hosts application workloads and control plane components, and a hosted cluster
+running on ARM cores of NVIDIA BlueField DPUs dedicated to networking functions
+and infrastructure services. The hosted cluster runs on extremely
+resource-constrained hardware that must accommodate all infrastructure services
+including networking, storage offloading, and security functions. 
+
+<details>
+<summary>DPF deployment overview</summary>
+
+  ![DPF deployment overview diagram](./assets/optional-monitoring-capability-doca.png)
+
+</details>
+
+DPUs exist specifically to offload infrastructure overhead from host x86
+servers, achieving up to 70% CPU savings on the host. Running full monitoring
+stacks on the DPU ARM cores would defeat this purpose by consuming the limited
+resources meant for high-performance networking and infrastructure services.
```

Comment:
"defeat" is a strong statement IMHO. From our conversations, I have the feeling that the current architecture has limited resources for workloads other than pure networking functions which drives a need for limiting the resource usage of platform components (especially with hosted control planes).

The consequence for this proposal is that when both RH telemetry *and* the capability are disabled, CMO shouldn't deploy Prometheus at all.